packages/connectivity/src/scp-cf/cache.ts

```diff
@@ -1,18 +1,30 @@
+import { Destination } from './destination';
+
 interface CacheInterface<T> {
-  hasKey(key: string): boolean;
-  get(key: string): T | undefined;
-  set(key: string, item: T, expirationTime?: number): void;
+  get(key: string | undefined): Promise<T | undefined>;
+  set(key: string | undefined, item: CacheEntry<T>): Promise<void>;
+  clear(): Promise<void>;
+}
+
+/**
+ * Interface to implement custom destination caching.
```

Comment:
[nth] Perhaps to can create a link to the `setDestinationCache` method with something like: Use the method setDestinationCache to set the custom implementation.